docs/tutorials/broadcast-join.rst

```diff
@@ -0,0 +1,298 @@
+Federal Information Processing Standard Publication codes that uniquely identify counties in the United States.
+
+If we make the assumption that a county with more GDELT events is somehow more politically relevant, we may be interested
+in seeing which counties are more dense with GDELT events.
+
+Since GDELT is a data set with point geometries, we do not immediately know which county it belongs to. To resolve this,
+we will use a second data set, a FIPS Codes shapefile that denotes the boundary and FIPS code of a county, and join GDELT
+points with the county that contains them.
+
+In this case, the number of counties is particularly small, approximately 3000 records, we can make our query much
+more efficient by "broadcasting" the counties. "Broadcast" here meaning a
+`Spark Broadcast <https://spark.apache.org/docs/2.2.0/api/java/org/apache/spark/sql/functions.html#broadcast-org.apache.spark.sql.Dataset->`__.
+In a traditional Spark SQL join, data will be shuffled around the executors based on the partitioners of the RDDs,
+and since in our case the join key is a geometric field, there is no Spark partitioner that can map data effectively.
+The resulting movement of data across nodes is expensive, so we can attain a performance boost by sending (broadcasting)
+our entire small data set to each of the nodes once. This ensures that the executors have all the data needed to compute
+the join, and no additional shuffling is needed.
+
+Spark provides a means of executing this "broadcast join", though it should only be used when the data being broadcast
+is small enough to fit in the memory of the executors.
+
+Prerequisites
+-------------
+
+For this tutorial, we will assume that your have already ingested the two data sets into the data store of your choosing.
+Following this tutorial without having created the necessary tables will lead to errors.
+
+If you have not ingested any data, you can follow one of the ingest tutorials :doc:`/tutorials/geomesa-examples-gdelt`.
+The converter for the GDELT data set, :doc:`/user/convert/premade/gdelt`, is provided with GeoMesa, and the FIPS data can
+be ingested without a converter as a shapefile.
+
+Initializing Spark
+------------------
+
+To start working with Spark, we will need a Spark Session initialized, and to apply GeoMesa's geospatial User Defined
+Types (UDTs) and User Defined Functions (UDFs) to our data in Spark, we will need to initialize our SparkSQL extensions.
+This functionality requires having the appropriate GeoMesa Spark runtime jar on the classpath when running your Spark job.
+GeoMesa provides spark runtime jars for Accumulo, HBase, and FileSystem data stores. For example, the following would start an
+interactive Spark REPL with all dependencies needed for running Spark with GeoMesa version 2.0.0 on an Accumulo data store.
+
+.. code-block:: bash
+
+    $ bin/spark-shell --jars geomesa-accumulo-spark-runtime_2.11-2.0.0.jar
```

Comment:
@mdzimmerman do we have any clear way to filter the GeoMesa version into the docs?

Reply:
my understanding is that we can't substitute in code blocks

Reply:
Yeah, you can use ``|release|`` and ``|version|`` in many places in the RST, but as Emilio states it doesn't work in code blocks.